### PR TITLE
[Ray 2.7 Examples][3/n] Delete Obsolete Transformers GLUE Example

### PR DESCRIPTION
**Pull request by woshiyyya**
## Why are these changes needed?

We clean up this outdated example which uses `AccelerateTrainer` API.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/train/examples/transformers/transformers_example.rst
Comment (matthewdeng) [inline doc/source/train/examples/transformers/transformers_example.rst:1]:
```
/home/docs/checkouts/readthedocs.org/user_builds/anyscale-ray/checkouts/38682/doc/source/train/examples.rst:: WARNING: toctree contains reference to nonexisting document 'train/examples/transformers/transformers_example' [etoc.ref]
```

**Review by matthewdeng — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/ToeiAnimation/">Toei Animation</a> on <a href="https://giphy.com/gifs/TOEIAnimationUK-one-piece-luffy-ace-death-THD4qcYQFew8IoE88b">GIPHY</a></div>

cc @GokuMohandas @angelinalg

File: doc/source/train/examples/transformers/transformers_example.rst
Comment (GokuMohandas) [reply doc/source/train/examples/transformers/transformers_example.rst:1]: Brutal PR by @woshiyyya but a necessary one. I plead everyone to keep this one alive in our memories at least.

Comment (woshiyyya) [reply doc/source/train/examples/transformers/transformers_example.rst:1]: Seems that I buried a memorable history here 😹 We'll remember it.

File: python/ray/train/BUILD
Comment (matthewdeng) [inline python/ray/train/BUILD:1]: @krfricke we're removing this example, which seems to be the only one included in the new persistence mode CI tests right now. 

Can you help determine the right next step for this (maybe disable the test suite until more tests are migrated) and help merge this PR after? Thanks!

**Review by krfricke — APPROVED** [review]

File: python/ray/train/BUILD
Comment (woshiyyya) [reply python/ray/train/BUILD:1]: ok sure. I'll add a `TODO()` and tag you there~

**Pull request merged**
